This is not a good way how to do it, we should rather emulate dvd for big isos, but it should work for data isos.
diff --git a/drivers/usb/gadget/function/storage_common.c b/drivers/usb/gadget/function/storage_common.c
index f7e6c42558eb..8db8cda0c930 100644
--- a/drivers/usb/gadget/function/storage_common.c
+++ b/drivers/usb/gadget/function/storage_common.c
@@ -243,12 +243,6 @@ int fsg_lun_open(struct fsg_lun *curlun, const char *filename)
        min_sectors = 1;
        if (curlun->cdrom) {
                min_sectors = 300;      /* Smallest track is 300 frames */
-               if (num_sectors >= 256*60*75) {
-                       num_sectors = 256*60*75 - 1;
-                       LINFO(curlun, "file too big: %s\n", filename);
-                       LINFO(curlun, "using only first %d blocks\n",
-                                       (int) num_sectors);
-               }
        }
        if (num_sectors < min_sectors) {
                LINFO(curlun, "file too small: %s\n", filename);